src/withRenew.js

```diff
@@ -263,25 +263,32 @@ const withRenew = WrappedComponent => class WithRenewComponent extends React.Com
       return;
     }
 
-    const q = loans.map(loan => loan.itemId).join(' or ');
-    const query = `(itemId==(${q})) and status==("Open - Awaiting pickup" or "Open - Not yet filled") sortby requestDate desc`;
-
-    reset();
-
-    GET({ params: { query } })
-      .then((requestRecords) => {
-        const requestCountObject = requestRecords.reduce((map, record) => {
-          map[record.itemId] = map[record.itemId]
-            ? map[record.itemId] + 1
-            : 1;
-
-          return map;
-        }, {});
-        this.setState({ requestCounts: requestCountObject });
-      });
+    // step through the loans list in small batches in order to create a
+    // short-enough query string that we can avoid a "414 Request URI Too Long"
+    // response from Okapi. Details at CHAL-30
+    const step = 50;
+    for (let i = 0; i < loans.length; i += step) {
+      const loansSlice = loans.slice(i, i + step);
+      const q = loansSlice.map(loan => loan.itemId).join(' or ');
+      const query = `(itemId==(${q})) and status==("Open - Awaiting pickup" or "Open - Not yet filled") sortby requestDate desc`;
+      reset();
+      GET({ params: { query } })
```

Comment:
In the current implementation, it is possible that request from later step will be returned first from the server. If the order does not matter than it is OK.

Reply:
The order doesn't matter since we're generating a map from `item.id` to request count. Do you think an implementation with Promise.all() would be cleaner/more clear? This was implemented along the lines of "I wonder if I can get this working in five minutes..." rather than "Let this code be a shining example of Good Design..."

Reply:
Sounds good.